sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -689,6 +689,18 @@ class SparkSqlAstBuilder extends AstBuilder {
         throw QueryParsingErrors.createFuncWithBothIfNotExistsAndReplaceError(ctx)
       }
 
+      val containsGeneratedColumn: Option[Boolean] = Option(ctx.parameters).map { params =>
+        params.colDefinition().asScala.exists { colDefinition =>
+          colDefinition.colDefinitionOption().asScala.exists { option =>
+            Option(option.generationExpression()).isDefined
```

Comment:
how about identity column?

Reply:
It also uses the generationExpression rule so it's rejected by this code correctly